server/services/matchTypeDecorator.ts

```diff
@@ -42,5 +42,5 @@ export class MatchTypeDecorator {
   }
 
   private isCourtReturn = <T extends ArrivalInfo>(item: T): boolean =>
-    item.isCurrentPrisoner && item.fromLocationType === 'COURT'
+    item.potentialMatches[0].arrivalType === 'FROM_COURT'
```

Comment:
this looks ropey but this is only called if a single potential match and `FROM_COURT` is type checked from the union